subprojects/core-api/src/main/java/org/gradle/api/artifacts/repositories/RepositoryContentDescriptor.java

```diff
@@ -0,0 +1,156 @@
+     *
+     * @param group the group name
+     * @param moduleName the module name
+     * @param version the module version
+     */
+    void includeVersion(String group, String moduleName, String version);
+
+    /**
+     * Declares that a specific module version should be searched for in this repository, using regular expressions.
+     *
+     * @param groupRegex the group name regular expression
+     * @param moduleNameRegex the module name regular expression
+     * @param versionRegex the module version regular expression
+     */
+    void includeVersionByRegex(String groupRegex, String moduleNameRegex, String versionRegex);
+
+    /**
+     * Declares that an entire group shouldn't be searched for in this repository.
+     *
+     * @param group the group name
+     */
+    void excludeGroup(String group);
+
+    /**
+     * Declares that an entire group shouldn't be searched for in this repository.
+     *
+     * @param groupRegex the group name regular expression
+     */
+    void excludeGroupByRegex(String groupRegex);
+
+    /**
+     * Declares that an entire module shouldn't be searched for in this repository.
+     *
+     * @param group the group name
+     * @param moduleName the module name
+     */
+    void excludeModule(String group, String moduleName);
+
+    /**
+     * Declares that an entire module shouldn't be searched for in this repository, using regular expressions.
+     *
+     * @param groupRegex the group name regular expression
+     * @param moduleNameRegex the module name regular expression
+     */
+    void excludeModuleByRegex(String groupRegex, String moduleNameRegex);
+
+    /**
+     * Declares that a specific module version shouldn't be searched for in this repository.
+     *
+     * @param group the group name
+     * @param moduleName the module name
+     * @param version the module version
+     */
+    void excludeVersion(String group, String moduleName, String version);
+
+    /**
+     * Declares that a specific module version shouldn't be searched for in this repository, using regular expressions.
+     *
+     * @param groupRegex the group name
+     * @param moduleNameRegex the module name
+     * @param versionRegex the module version
+     */
+    void excludeVersionByRegex(String groupRegex, String moduleNameRegex, String versionRegex);
+
+    /**
+     * Declares that this repository should only be used for a specific
+     * set of configurations. Defaults to any configuration
+     *
+     * @param configurationNames the names of the configurations the repository will be used for
+     */
+    void onlyForConfigurations(String... configurationNames);
+
+    /**
+     * Declares that this repository should not be used for a specific
+     * set of configurations. Defaults to any configuration
+     *
+     * @param configurationNames the names of the configurations the repository will not be used for
+     */
+    void excludeConfigurations(String... configurationNames);
```

Comment:
:o: `notForConfigurations()` - to be consistent with the `onlyForConfigurations()` method and not to be mixed up with the `exclude...()` methods.